core/src/main/java/org/pentaho/platform/web/http/api/resources/services/ISchedulerServicePlugin.java

```diff
@@ -0,0 +1,111 @@
+/*!
+ * HITACHI VANTARA PROPRIETARY AND CONFIDENTIAL
+ *
+ * Copyright 2023 Hitachi Vantara. All rights reserved.
+ *
+ * NOTICE: All information including source code contained herein is, and
+ * remains the sole property of Hitachi Vantara and its licensors. The intellectual
+ * and technical concepts contained herein are proprietary and confidential
+ * to, and are trade secrets of Hitachi Vantara and may be covered by U.S. and foreign
+ * patents, or patents in process, and are protected by trade secret and
+ * copyright laws. The receipt or possession of this source code and/or related
+ * information does not convey or imply any rights to reproduce, disclose or
+ * distribute its contents, or to manufacture, use, or sell anything that it
+ * may describe, in whole or in part. Any reproduction, modification, distribution,
+ * or public display of this information without the express written authorization
+ * from Hitachi Vantara is strictly prohibited and in violation of applicable laws and
+ * international treaties. Access to the source code contained herein is strictly
+ * prohibited to anyone except those individuals and entities who have executed
+ * confidentiality and non-disclosure agreements or other agreements with Hitachi Vantara,
+ * explicitly covering such access.
+ */
+
+package org.pentaho.platform.web.http.api.resources.services;
+
+import org.pentaho.platform.api.repository2.unified.webservices.RepositoryFileDto;
+import org.pentaho.platform.api.scheduler2.IJob;
+import org.pentaho.platform.api.scheduler2.IJobTrigger;
+import org.pentaho.platform.api.scheduler2.IScheduler;
+import org.pentaho.platform.api.scheduler2.Job;
+import org.pentaho.platform.api.scheduler2.JobState;
+import org.pentaho.platform.api.scheduler2.SchedulerException;
+import org.pentaho.platform.web.http.api.proxies.BlockStatusProxy;
+import org.pentaho.platform.web.http.api.resources.JobRequest;
+import org.pentaho.platform.web.http.api.resources.JobScheduleRequest;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Interface for pentaho platform scheduling service with focus
+ * on servicing RESTful Web Services and real implementation of classes.
+ * Ideally a more generic interface moved to pentaho-platform/scheduler that
+ * only has interfaces for method arguments and return objects.
+ */
+/*
+ * TODO look into refactoring SchedulerService to use *only* interfaces for arguments/return objects like:
+ *  - pentaho-platform/scheduler/src/main/java/org/pentaho/platform/api/scheduler2/IJob.java
+ * so this class can be moved to:
+ *  - pentaho-platform/scheduler/src/main/java/org/pentaho/platform/api/scheduler2/ISchedulerService.java
+ */
+public interface ISchedulerServicePlugin {
```

Comment:
Not sure about this `Plugin` suffix on the name.

Reply:
My first and obvious choice for an interface name was `ISchedulerService.java`

At the time, and some there are two other similar named classes. Yes. they have different packages.

- https://github.com/pentaho/pentaho-platform/tree/scheduler-plugin/scheduler/src/main/java/org/pentaho/platform/api/scheduler2
- https://github.com/pentaho/pdi-scheduler-plugin/blob/master/core/src/main/java/org/pentaho/di/spoon/scheduler/ws/SchedulerServiceFactory.java

I made a decision to append "Plugin" to denote that is it is infact in the https://github.com/pentaho/pentaho-scheduler-plugin project to avoid any confusion.

What would you suggest, if any changes?

Reply:
Oh, no, let it be. I guessed there would be a reason for the `Plugin` name. Just wanted to confirm.